src/components/deploymentConfig/DeploymentConfig.tsx

```diff
@@ -58,10 +63,19 @@ export default function DeploymentConfig({
     environments,
     isProtected,
     reloadEnvironments,
+    isSuperAdmin
 }: DeploymentConfigProps) {
     const history = useHistory()
     const { appId } = useParams<{ appId: string }>()
     const { currentServerInfo } = useContext(mainContext)
+    const [saveChangesCheckbox, setSaveChangesCheckbox] = useState(false)
+    const [showLockedDiffForApproval, setShowLockedDiffForApproval] = useState(false)
+    const [lockedKeysArray, setLockedKeysArray] = useState<{ config: string[]; allowed: boolean }>({
```

Comment:
extract this type out to a single place and import it, as it is used at multiple places:
{ config: string[]; allowed: boolean }

Reply:
done